### PR TITLE
#1132 Unify training data selection for evaluation

### PR DESCRIPTION
**Pull request by UWinch**
During evaluation, different recommenders select different amounts of training data given the same data and the same DataSplitter. To make recommender evaluation more comparable, the used training data is now given as ratio compared to all training data of the selected recommender for the evaluation simulation.

**What's in the PR**
* Implemented ratio of used training data as field in the evaluation result.
* Also added a message field to the result object to communicate possible errors to the user via the learning curve chart.

**How to test manually**
* Configure a project with a recommender and annotated data.
* Start an evaluation simulation on the recommender tab in the project settings.
* Check that the learning curve on the annotation page is still properly shown.

**Unrelated Changes**
* Update to admin guide regarding the learning.curve flag.

### CODE REVIEW

**Comment by reckart** [comment]
> Set the learning curve to be enabled via the project.settings

I believe the learning curve is already "always enabled" and cannot even be turned off anymore.

**Comment by UWinch** [comment]
You are right, the flag for the learning curve is gone. I'll use this issue to update the admin docu on this.

File: inception-imls-dl4j/src/main/java/de/tudarmstadt/ukp/inception/recommendation/imls/dl4j/pos/DL4JSequenceRecommender.java [hunk 1]
```diff
@@ -493,12 +493,17 @@ public EvaluationResult evaluate(List<CAS> aCas, DataSplitter aDataSplitter)
 
         int testSetSize = testSet.size();
         int trainingSetSize = trainingSet.size();
+        double trainRatio = (double) trainingSetSize / (double) (data.size() - testSetSize);
         
         if (trainingSetSize < 2 || testSetSize < 2) {
-            log.info("Not enough data to evaluate, skipping!");
+            String info = String.format(
+                    "Not enough training data: training set [%s] items, test set [%s] of total [%s].",
+                    trainingSetSize, testSetSize, data.size());
```
Comment (jcklie) [inline inception-imls-dl4j/src/main/java/de/tudarmstadt/ukp/inception/recommendation/imls/dl4j/pos/DL4JSequenceRecommender.java]: You can directly log that with log.info and "{}" as place holders, no need for String.format.

File: inception-recommendation-api/src/main/java/de/tudarmstadt/ukp/inception/recommendation/api/evaluation/EvaluationResult.java [hunk 2]
```diff
@@ -36,7 +36,13 @@
 {
     private final int trainingSetSize;
     private final int testSetSize;
+    
+    /**
+     * Rate of this training data compared to all training data
+     */
+    private final double trainingDataRatio;
     private boolean skippedEvaluation;
+    private String errorMsg;
```
Comment (jcklie) [inline inception-recommendation-api/src/main/java/de/tudarmstadt/ukp/inception/recommendation/api/evaluation/EvaluationResult.java]: I would like that String to be Optional

Comment (reckart) [reply inception-recommendation-api/src/main/java/de/tudarmstadt/ukp/inception/recommendation/api/evaluation/EvaluationResult.java]: I think we should avoid `Optional` fields. If it makes sense to make the getter optional, then return `Optional.ofNullable(...)`.

**Comment by jcklie** [comment]
When I use the Example Recommender Project, then I do not see a graph in the coordinate system and the x-axis is from -1 to 1.

**Comment by UWinch** [comment]
Thx, I will do some debugging.

**Pull request merged**